### PR TITLE
Ticket1446: Don't push configs onto the wrong branch

### PR DESCRIPTION
**Pull request by DominicOram**
### Description of work

See https://github.com/ISISComputingGroup/IBEX/issues/1446

Version control is only started when on the same branch as your host name 
### To test
- Start an instrument with your configurations under git version control, confirm that any modifications/additions made through the blockserver are added to the remote repo
- Stop the instrument, switch your configurations to another branch
- Start the instrument and confirm that no config changes are pushed automatically when modifications are made through the blockserver

---
#### Code Review
- [x] Is the code of an acceptable quality?
- [x] Has the author taken into account the multi-thredded nature of the code?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
### Functional Tests
- [x] Do changes function as described? Add comments below that describe the tests performed.
### Final steps
- [x] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

**Comment by IsabellaRey** [comment]
These are the tests I've done as part of the review:

* With the instrument not running, create a branch with name equal to my fedid: `git checkout -b myfedid`
* Start the instrument
* Any changes are not even added locally anymore
- [ ] Is the above behaviour what we want?

**Comment by IsabellaRey** [comment]
P.S. Could you please also add an entry in the manual system tests spreadsheet to check that rcptt files are not being considered for version control

**Comment by DominicOram** [comment]
Fixed issues, expected behaviour is now outlined in https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/BlockServer#the-config-version-control

File: ConfigVersionControl/config_version_control.py [hunk 1]
```diff
@@ -11,9 +15,20 @@ def __init__(self, working_directory):
     #i.e. not automatically via the file observer firing an event, but after a new configuration is created
```
Comment (mattclarke) [inline ConfigVersionControl/config_version_control.py]: hash then space then comment - can you fix these even though you didn't create them please?

Comment (mattclarke) [inline ConfigVersionControl/config_version_control.py]: Looking closer: what are these comments related to? If it is a method then they should be inside it

**Comment by IsabellaRey** [comment]
I have approved changes in case someone else needs to pick this up for review.

Regarding the changes I requested: I am happy with those related to documentation and release notes.

There are still a couple of issues:
- [ ] git_version_control.py needs an "import os" otherwise it throws an exception and nothing is committed to the remote repo, because the blockserver ends up using the Mock Version Control. Can you please add this

With the import os added, I have tested the following:
* branch with name equal to my machine name pushed changes to remote repo --> OK
* branch with name equal to my fedid pushed changes to remote repo -->OK
* branch with name NDISABELLA_TEST did not push --> OK
* master branch did not push --> OK

One outstanding issue:
- [ ] config and synoptic with name starting "rcptt_" still got pushed remotely (tested on branch whoe name was my fedid), even though the check for the rcptt prefix is in the code

NOTE TO NEXT REVIEWER:
I haven't had time to look at the actual code, I've only tested the functionality

**Comment by DominicOram** [comment]
Fixed import error. If there are pre-existing rcptt configs these will get pushed. However, any new configs created with rcptt prefix will not be added.
